### ISSUE TITLE
[selectors-4] misleading/incomplete Note: clarify that <forgiving-selector-list> does not work as expected with pseudo-elements

### ISSUE THREAD

**Issue by claudepache** [w3c/csswg-drafts#6792]
The Note in Section [17.1 \<forgiving-selector-list\> and \<forgiving-relative-selector-list\>](https://drafts.csswg.org/selectors-4/#forgiving-selector) says:

> Note: Style rules still use the normal, unforgiving selector list behavior. \<forgiving-selector-list\> is used in some functions, like :is(), which are similarly generic. Although it does have some minor implications on specificity, simply wrapping a style rule’s selector in :is() effectively "upgrades" it to become forgiving.

There is however a much more serious implication than specificity when the selector list targets pseudo-elements. The following won’t work as naïvely expected ([js-bin live example](https://jsbin.com/mizudepuga/edit?html,css,output)):

```css
:is(button, input[type=button], input[type=submit], ::file-selector-button) {
    background: fuchsia;
}
```

**Comment by tabatkins** [comment]
The "implication" is just that pseudo-elements aren't valid in `:is()`, right?

**Comment by Loirooriol** [comment]
That's already implied in the spec:

> Pseudo-elements cannot be represented by the matches-any pseudo-class; they are not valid within :is().

#2284 is about making them valid (deferred to L5)

Forbidding pseudo-elements in forgiving selector lists will also cover `:has()`, which seems good.

**Comment by claudepache** [comment]
Just for clarity. Both specificity and pseudo-element handling of `:is()` are correctly mentioned in the spec.  My issue is that (unless #2284 is fixed) the specific Note in section 17.1 is currently misleading, because developers *will* try to wrap some selector list containing some partially supported pseudo-element with `:is()`.

**Comment by tabatkins** [comment]
All right, text amended to clarify that you can't "upgrade" a selector list that contains pseudo-elements.
